### ISSUE TITLE
"browsingContext.captureScreenshot" should support full page screenshots

### ISSUE THREAD

**Issue by whimboo** [w3c/webdriver-bidi#384]
Right now the command only takes a screenshot from the visible viewport. We should extend the command to also capture full page screenshots.

**Comment by mathiasbynens** [comment]
Is this the same as taking an element screenshot (`ElementClipRectangle`) of the root element?

**Comment by mathiasbynens** [comment]
Is this the same as taking an element screenshot (`ElementClipRectangle`) of the root element? If not, what are the differences?

**Comment by jgraham** [comment]
No, because even with element screenshots we currently clip to the viewport. Conceptually full-screen screenshots require us to generate a new viewport with dimensions equal to the scroll width and scroll height of the document, render to that viewport, and then (for full screen / element screenshots) clip against that.

**Comment by sadym-chromium** [comment]
I guess we are got to consensus that the screenshots of things beyond viewports should be possible. The questions are:
* Do we have scenarios where clipping a screenshot of a box (or element) to the viewport is required? 
* Or if it is required only for backward compatibility with WebDriver Classic, would it be acceptable to be done in 2 calls (with potential inconsistency)?

In other words, can we disrespect viewport during taking screenshot unless it's a viewport's screenshot? If so, the following API can be done:

```
browsingContext.CaptureScreenshotParameters = {
  context: browsingContext.BrowsingContext,
  ? rectangle: browsingContext.Rectangle // // CHANGED HERE: Defaults to ViewportRectangle
}

browsingContext.Rectangle = (
 browsingContext.ViewportClipRectangle / // CHANGED HERE
 browsingContext.FullpageClipRectangle / // CHANGED HERE
  browsingContext.BoxClipRectangle /
  browsingContext.ElementClipRectangle
)

browsingContext.ViewportClipRectangle = {  // CHANGED HERE
  type: "viewport"
}

browsingContext.FullpageClipRectangle = {  // CHANGED HERE
  type: "fullpage"
}

browsingContext.ElementClipRectangle = {
  type: "element",
  element: script.SharedReference,
  ? scrollIntoView: bool
}

browsingContext.BoxClipRectangle = {
   type: "box", // CHANGED HERE. Renamed from "viewport" to "box" to avoid confusion.
   x: float,
   y: float,
   width: float,
   height: float
}
```

**Comment by whimboo** [comment]
This was fixed by PR #564.
